### PR TITLE
fix(moltype): improve moltype validation

### PR DESCRIPTION
**Pull request by Qinzi27**
### Summary
Fix moltype validation logic to handle edge cases correctly.

### Changes
- Improved moltype validation to check for unexpected input values.
- Updated related test cases.

### Reason
Previously, the validation did not handle some special cases, which could lead to incorrect classification.

### Impact
- Affects `moltype` validation module.
- No breaking changes expected.

### Test
- Added new test cases to cover edge cases.
- Manually tested with various moltype inputs.

## Summary by Sourcery

Improves moltype validation to check for unexpected input values and raises an error if the moltype is invalid.

Bug Fixes:
- Fixes an issue where the moltype validation logic did not handle unexpected input values, leading to incorrect classification.

Tests:
- Adds a new test case to cover the edge case of invalid moltype inputs.

### CODE REVIEW

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
Thanks for the effort! Please see my suggestion regarding the error message change. You are super close to getting this PR accepted!!

**Comment by GavinHuttley** [comment]
@Qinzi27 please also run the linter command in your terminal before pushing in your next revisions, i.e.
```
$ cd to/repo/root
$ ruff check --fix-only . && ruff format .
```

**Comment by Qinzi27** [comment]
> ruff check --fix-only . && ruff format .

It gives me those warning.
_Fixed 1 error (29 additional fixes available with `--unsafe-fixes`).
warning: The following rule may cause conflicts when used with the formatter: `COM812`. To avoid unexpected behavior, we recommend disabling this rule, either by removing it from the `select` or `extend-select` configuration, or adding it to the `ignore` configuration.
2 files reformatted, 310 files left unchanged_

File: src/cogent3/app/sample.py [hunk 1]
```diff
@@ -970,8 +970,10 @@ def __init__(
             moltype = cogent3.get_moltype(moltype)
         valid_moltypes = {"dna", "rna", "protein", "protein_with_stop"}
         if moltype.label.lower() not in valid_moltypes:
-            raise new_moltype.MolTypeError(f"Invalid moltype: {moltype.label}. Moltype must be one of DNA, RNA, PROTEIN, or PROTEIN WITH STOP.")
-        
+            raise new_moltype.MolTypeError(
+                f"Invalid moltype: {moltype.label}. Moltype must be one of DNA, RNA, PROTEIN, or PROTEIN WITH STOP."
```
Comment (GavinHuttley) [inline src/cogent3/app/sample.py:974]: change "PROTEIN WITH STOP" to "PROTEIN_WITH_STOP"

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
see comment, so close!

File: tests/test_app/test_sample.py [hunk 2]
```diff
@@ -499,6 +500,14 @@ def test_omit_bad_seqs(bad_gap_data, gap_fraction, quantile, expected_keys):
     assert got.to_dict() == expected
 
 
+def test_omit_bad_seqs_error():
+    with pytest.raises(
+        MolTypeError,
+        match="Invalid moltype: text. Moltype must be one of DNA, RNA, or PROTEIN.",
```
Comment (GavinHuttley) [inline tests/test_app/test_sample.py:506]: This test is not passing because the error messages do not match. I know that the AI tools recommend checking the error messages, but I disagree. I think they make the test fragile, as this case shows. Delete the match line.

**Review by GavinHuttley — CHANGES_REQUESTED** [review]
The new test fails ☹️ (look under the "checks" section for details). In general, it's a good idea to run the tests on your own machine before pushing.

Simple fix suggested.

**Comment by GavinHuttley** [comment]
@Qinzi27 checking you've seen that the test suite is failing?

**Review by GavinHuttley — APPROVED** [review]
Awesome, great job @Qinzi27 !
